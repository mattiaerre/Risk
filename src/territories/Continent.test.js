const Continent = require('./Continent');

test('Australia', () => {
  // eslint-disable-next-line global-require
  const territories = require('../data/australia.json');

  const australia = new Continent({ id: 'australia', name: 'Australia' });

  expect(australia).toMatchSnapshot();

  australia.makeTerritories(territories);

  expect(australia.territories.length).toBe(4);

  expect(
    australia.territories.map((territory) => territory.name)
  ).toMatchSnapshot();
});

test('North America', () => {
  // eslint-disable-next-line global-require
  const territories = require('../data/north-america.json');

  const northAmerica = new Continent({
    id: 'north-america',
    name: 'North America'
  });

  expect(northAmerica).toMatchSnapshot();

  northAmerica.makeTerritories(territories);

  expect(northAmerica.territories.length).toBe(9);

  expect(
    northAmerica.territories.map((territory) => territory.name)
  ).toMatchSnapshot();

  const westernUnitedStates = northAmerica.territories.find(
    (territory) => territory.id === 'western-united-states'
  );

  expect(westernUnitedStates.bordersWith('eastern-united-states')).toBe(true);

  const alaska = northAmerica.territories.find(
    (territory) => territory.id === 'alaska'
  );

  // INFO: this invalidates my design so far :(
  expect(alaska.bordersWith('kamchatka')).toBe(true);
});
