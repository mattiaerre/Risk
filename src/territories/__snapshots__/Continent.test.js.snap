// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Australia 1`] = `
Continent {
  "id": "australia",
  "name": "Australia",
  "territories": Array [],
}
`;

exports[`Australia 2`] = `
Array [
  "Eastern Australia",
  "Indonesia",
  "New Guinea",
  "Western Australia",
]
`;

exports[`North America 1`] = `
Continent {
  "id": "north-america",
  "name": "North America",
  "territories": Array [],
}
`;

exports[`North America 2`] = `
Array [
  "Alaska",
  "Alberta (Western Canada)",
  "Central America",
  "Eastern United States",
  "Greenland",
  "Northwest Territory",
  "Ontario (Central Canada)",
  "Quebec (Eastern Canada)",
  "Western United States",
]
`;
